internal/scorecard/tests/olm.go

```diff
@@ -335,7 +343,10 @@ func isCRFromCRDApi(cr unstructured.Unstructured, crds []*apiextv1.CustomResourc
 				}
 			}
 			if cr.Object["status"] != nil {
-				status := cr.Object["status"].(map[string]interface{})
+				status, ok := cr.Object["status"].(map[string]interface{})
+				if !ok {
+					failed = true
+				}
 				for key := range status {
 					if _, ok := version.Schema.OpenAPIV3Schema.Properties["status"].Properties[key]; !ok {
 						failed = true
```

Comment:
If you apply the same checks as suggested no changes should be required. 
Then you can do another pr to enable any other check that you would like with the fixes.

Reply:
@camilamacedo86 I'm not sure what you want to do here... The previous golangci-lint was version `v1.27.0`.  If I switch back to that, there are 3 "errcheck" failures in this file.  I do not understand why this doesn't come up in the CI or on others machines, but these are valid errcheck failures for which I've provided a fix.  Based on no differences captured in linting between `v1.27.0` and `v.1.30.0`, it seemed reasonable to update to the latest.

At this point:
* we can go with better code that doesn't ignore error checks
* I can remove the checks and add annotations to ignore these as err checks
* we can close this PR and continue to change developers global toolchain 

open to other suggests as well...

Reply:
We should have a PR with the change that starts to raise the issue and the fixes for that only. Otherwise, it is hard we verify that. It might be a new check that and then, we should discuss if the new check should or not be added, If the new check is beneficial or not, as if the changes/fixes are accurate or not and etc..